controllers/clusterclaims.go

```diff
@@ -0,0 +1,145 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	operatorsv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strings"
+
+	"github.com/go-logr/logr"
+	odfv1alpha1 "github.com/red-hat-storage/odf-operator/api/v1alpha1"
+	extensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	clusterv1alpha1 "open-cluster-management.io/api/cluster/v1alpha1"
+)
+
+type ClusterClaimKey string
+
+const (
+	OdfVersion            ClusterClaimKey = "odf-version"
```

Comment:
What do you think about adding odf as a prefix in front of all clusterClaim names? I feel it will avoid name conflict and safer to add these names in ACM Managed cluster

Reply:
It will help us identify that it belongs to odf but i don't think there will be name conflict for clusterclaim. 
For identification, i can add owner ref to the storagesystem and it will also help us with the uninstall. What do you think ?

Reply:
when you create cluster claim on odf cluster, it will create one status entry on ACM managed cluster CR, i think i will use the same which you used for cluster claim. so when it displayed on ACM  managed cluster CR, we need to makesure our ODF names dont have a conflict with others.

Reply:
ya the owner ref is another good feature to cleanup